### ISSUE TITLE
Compose Multiplatform 1.5.0 gradle detected problems

### ISSUE THREAD

**Issue by Nailik** [icerockdev/moko-resources#530]
Fix:

```
tasks.withType<SyncComposeResourcesForIosTask> {
    dependsOn(tasks.findByName("generateMRcommonMain"))
    dependsOn(tasks.findByName("generateMRiosSimulatorArm64Main"))
    dependsOn(tasks.findByName("generateMRiosArm64Main"))
}
```

Versions:
Kotlin: 1.9.0-RC
moko..resources=0.23.0
gradle gradle-8.2.1-all

Issue log:

```

Showing All Messages
> Task :resources:syncPodComposeResourcesForIos FAILED



FAILURE: Build failed with an exception.



* What went wrong:

Some problems were found with the configuration of task ':resources:syncPodComposeResourcesForIos' (type 'SyncComposeResourcesForIosTask').

  - Gradle detected a problem with the following location: '/Users/KIEller/rhasspy_mobile/resources/build/generated/moko/commonMain/res'.

    

    Reason: Task ':resources:syncPodComposeResourcesForIos' uses this output of task ':resources:generateMRcommonMain' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

    

    Possible solutions:

      1. Declare task ':resources:generateMRcommonMain' as an input of ':resources:syncPodComposeResourcesForIos'.

      2. Declare an explicit dependency on ':resources:generateMRcommonMain' from ':resources:syncPodComposeResourcesForIos' using Task#dependsOn.

      3. Declare an explicit dependency on ':resources:generateMRcommonMain' from ':resources:syncPodComposeResourcesForIos' using Task#mustRunAfter.

    

    For more information, please refer to https://docs.gradle.org/8.2.1/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.

  - Gradle detected a problem with the following location: '/Users/KIEller/rhasspy_mobile/resources/build/generated/moko/iosSimulatorArm64Main/res'.

    

    Reason: Task ':resources:syncPodComposeResourcesForIos' uses this output of task ':resources:generateMRiosSimulatorArm64Main' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

    

    Possible solutions:

      1. Declare task ':resources:generateMRiosSimulatorArm64Main' as an input of ':resources:syncPodComposeResourcesForIos'.

      2. Declare an explicit dependency on ':resources:generateMRiosSimulatorArm64Main' from ':resources:syncPodComposeResourcesForIos' using Task#dependsOn.

      3. Declare an explicit dependency on ':resources:generateMRiosSimulatorArm64Main' from ':resources:syncPodComposeResourcesForIos' using Task#mustRunAfter.

    

    For more information, please refer to https://docs.gradle.org/8.2.1/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.

  - Gradle detected a problem with the following location: '/Users/KIEller/rhasspy_mobile/resources/build/generated/moko/iosSimulatorArm64Main/src'.

    

    Reason: Task ':resources:syncPodComposeResourcesForIos' uses this output of task ':resources:generateMRiosSimulatorArm64Main' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

    

    Possible solutions:

      1. Declare task ':resources:generateMRiosSimulatorArm64Main' as an input of ':resources:syncPodComposeResourcesForIos'.

      2. Declare an explicit dependency on ':resources:generateMRiosSimulatorArm64Main' from ':resources:syncPodComposeResourcesForIos' using Task#dependsOn.

      3. Declare an explicit dependency on ':resources:generateMRiosSimulatorArm64Main' from ':resources:syncPodComposeResourcesForIos' using Task#mustRunAfter.

    

    For more information, please refer to https://docs.gradle.org/8.2.1/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.

```

**Comment by hichamboushaba** [comment]
I'm not sure the cause is related to Gradle version, but instead it's linked to the version of Compose Multiplatform, I believe the cause is this change to the Compose Multiplatform plugin:   https://github.com/JetBrains/compose-multiplatform/pull/3340

For anyone who doesn't Compose Resources component in addition to Moko, you can workaround the issue by adding the property: `org.jetbrains.compose.ios.resources.sync=false` to your gradle properties.

**Comment by marenovakovic** [comment]
> I'm not sure the cause is related to Gradle version, but instead it's linked to the version of Compose Multiplatform, I believe the cause is this change to the Compose Multiplatform plugin: [JetBrains/compose-multiplatform#3340](https://github.com/JetBrains/compose-multiplatform/pull/3340)
> 
> For anyone who doesn't use Compose Resources component in addition to Moko, you can workaround the issue by adding the property: `org.jetbrains.compose.ios.resources.sync=false` to your gradle properties.

What about us who use both `compose.resources` and `moko`?

**Comment by AlexeyTsvetkov** [comment]
FYI: we (Compose Multiplatform team) discussed the issue internally. So far our conclusion is that making different resource libraries working together would probably be complicated and fragile. As a workaround, the next release of Compose Multiplatform Gradle plugin will turn off resource management automatically when `moko-resources` or `libres` Gradle plugins are used.

**Comment by NakiriYuuzu** [comment]
waiting 15hour is that normal?
![image](https://github.com/icerockdev/moko-resources/assets/72718339/62ff5489-0e13-4843-ac8a-8d5f89d26b00)

**Comment by marenovakovic** [comment]
Thank you for the update.
For now I dropped `moko-resources` and I'm using string literals :/.
I will go back to `moko-resources` when this is fixed.

**Comment by ExNDY** [comment]
will be fixed in 0.24.0, please check in 0.24.0-alpha-6

**Comment by Alex009** [comment]
should be fixed in 0.24.0-beta-1
